code/game/objects/items/carp_lasso.dm

```diff
@@ -65,28 +82,51 @@
 
 /obj/item/mob_lasso/proc/handle_hard_del()
 	mob_target = null
+	timer = null
 
 ///Primal version, allows lavaland goobers to tame goliaths
 /obj/item/mob_lasso/primal
 	name = "primal lasso"
-	desc = "Found amongst Ash Walker tools.\nCan be used to tame goliaths."
-	whitelist_mobs = list(/mob/living/simple_animal/hostile/asteroid/goliath/beast, /mob/living/simple_animal/hostile/asteroid/goliath, /mob/living/simple_animal/hostile/asteroid/goldgrub)
+	desc = "A lasso fashioned out of goliath plating that is often found in the possession of Ash Walkers.\nCan be used to tame some lavaland animals."
+
+/obj/item/mob_lasso/primal/Initialize(mapload)
+	. = ..()
+	whitelist_mobs = typecacheof(list(/mob/living/simple_animal/hostile/asteroid/goliath, /mob/living/simple_animal/hostile/asteroid/goldgrub,\
+		/mob/living/simple_animal/hostile/asteroid/basilisk/watcher, /mob/living/simple_animal/hostile/asteroid/gutlunch))
+
+/obj/item/mob_lasso/drake
+	name = "drake lasso"
+	desc = "A lasso fashioned out of the scaly hide of an ash drake.\nCan be used to tame one, if you can get close enough."
+	range = 3
+
+/obj/item/mob_lasso/drake/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!user.mind?.has_antag_datum(/datum/antagonist/ashwalker))
+		to_chat(user, "<span class='warning'>You don't know how to use this!</span>")
+		return
+	. = ..()
+
+/obj/item/mob_lasso/drake/Initialize(mapload)
+	. = ..()
+	whitelist_mobs = typecacheof(list(/mob/living/simple_animal/hostile/megafauna/dragon), only_root_path = TRUE)
 
 /obj/item/mob_lasso/antag
 	name = "bluespace lasso"
 	desc = "Comes standard with every evil space-cowboy!\nCan be used to tame almost anything."
 	///blacklist of disallowed mobs
-	var/list/blacklist_mobs = list(/mob/living/simple_animal/hostile/megafauna, /mob/living/simple_animal/hostile/alien, /mob/living/simple_animal/hostile/syndicate)
+	var/list/blacklist_mobs
```

Comment:
I don't like the blacklist functionality being attached to a subtype for antags, the debug being a subtype of antag is weird.
Both vars should be on the parent type, or at minimum a blacklist type that debug and antag are both children of.

Antag should be renamed to traitor probably, I know that's not part of this PR

Reply:
Alright, moved it to the parent.